nomt/src/rw_pass_cell/mod.rs

```diff
@@ -383,20 +394,34 @@ impl<'a, 'pass, T> ReadGuard<'a, 'pass, T>
 where
     'pass: 'a,
 {
+    #[cfg(not(loom))]
     /// Get a reference to the underlying value.
     pub fn get(&self) -> &'pass T {
         // SAFETY: The existence of the guard ensures that there are only shared references
         //         to the inner value. The returned reference cannot outlive the guard.
         unsafe { &*self.inner.get() }
     }
+
+    #[cfg(loom)]
+    pub fn with<R>(&self, f: impl FnOnce(&T) -> R) -> R {
+        // SAFETY: The existence of the guard ensures that there is only one mutable reference
+        //         to the inner value. The returned reference cannot outlive the guard.
+        self.inner.with(|inner| f(unsafe { &*inner }))
+    }
 }
 
 impl<'a, 'pass, T> Deref for ReadGuard<'a, 'pass, T> {
     type Target = T;
 
+    #[cfg(not(loom))]
     fn deref(&self) -> &Self::Target {
         self.get()
     }
+
+    #[cfg(loom)]
+    fn deref(&self) -> &Self::Target {
+        unreachable!("with loom cfg, ReadGuard::deref cannot be used")
```

Comment:
Because inside the module `page_cache`, the `Deref` trait is expected to be implemented for both `WriteGuard` and `ReadGuard`, if we make the whole impl `cfg(not(loom))`, the `nomt` crate would not compile with `RUSTFLAG="--cfg loom"`.

This solution is super tricky, but I found it to be the cleanest one to avoid changing the access pattern to `Read/WriteGuard`. It works if and only if inside tests we make sure there is no auto deref coercion (or we use it explicitly)